tutorial/tests/testsuite.py

```diff
@@ -277,12 +149,42 @@ def ipytest(self, line: str, cell: str):
                     pytest_stdout.getvalue()
                     pytest_stderr.getvalue()
 
+                # Find the respective reference solution for the executed function
+                # Create a str containing its code and the code of all other functions used,
+                # whether coming from the same file or an imported one
+                solution_functions = [
+                    val
+                    for key, val in called_function_names.items()
+                    if key in f"reference_{name}"
+                ][0]
+                solution_code = ""
+
+                for f in solution_functions:
+                    if f in function_defs:
+                        solution_code += ast.unparse(function_defs[f]) + "\n\n"
+                    elif f in function_imports:
+                        function_file = pathlib.Path(
+                            f"{function_imports[f].replace('.', '/')}.py"
+                        )
+                        if function_file.exists():
+                            function_file_tree = ast.parse(open(function_file).read())
+                            for node in function_file_tree.body:
+                                if (
+                                    isinstance(
+                                        node, (ast.FunctionDef, ast.AsyncFunctionDef)
+                                    )
+                                    and node.name == f
+                                ):
+                                    solution_code += ast.unparse(node) + "\n\n"
+
                 outputs.append(
                     TestResultOutput(
                         name,
                         False,
                         result == pytest.ExitCode.OK,
                         list(result_collector.tests.values()),
+                        self.cells[cell_id],
+                        solution_code,
```

Comment:
Could you make Ipython display solution code as a code cell with syntax highlighting?

Reply:
This is already handled. Please see class TestResultOutput in testsuite_helpers.py